tests/integration/test_cli_service_accounts.py

```diff
@@ -181,26 +220,99 @@ def test_create_service_account(namespace, backend, action, resource, primary):
         expected_labels.update({PRIMARY_LABELNAME: "True"})
     assert actual_labels == expected_labels
 
-    # Check for RBAC permissions
-    sa_identifier = f"system:serviceaccount:{namespace}:{username}"
-    rbac_check = subprocess.run(
+    # Check secret creation
+    secret_result = subprocess.run(
         [
             "kubectl",
-            "auth",
-            "can-i",
-            action,
-            resource,
-            "--namespace",
+            "get",
+            "secret",
+            secret_name,
+            "-n",
             namespace,
-            "--as",
-            sa_identifier,
+            "-o",
+            "json",
         ],
         check=True,
         capture_output=True,
         text=True,
     )
-    assert rbac_check.returncode == 0
-    assert rbac_check.stdout.strip() == "yes"
+    assert secret_result.returncode == 0
+
+    # Check for RBAC permissions
+    sa_identifier = f"system:serviceaccount:{namespace}:{username}"
+    for resource, actions in ALLOWED_PERMISSIONS.items():
```

Comment:
**praise**: I think doing these tests inside the single context of the function would preserve a lot of time in preparing / destroying fixtures.